src/autorest-core/language-service/language-service.ts

```diff
@@ -0,0 +1,653 @@
+    this.clear();
+
+    // set configuration
+    await this.resetConfiguration(this.service.settings.configuration)
+
+    // get the list of files this is running on
+    this.files = (await this.AutoRest.view).InputFileUris;
+
+    // start it up!
+    const processResult = this.AutoRest.Process();
+
+    this.cancel = async () => {
+      // cancel only once!
+      this.cancel = async () => { };
+
+      // cancel the current process if running.
+      processResult.cancel();
+    };
+  }
+
+  public async resetConfiguration(configuration: any) {
+    // wipe the previous configuration
+    await this.AutoRest.ResetConfiguration();
+
+    // set the basic defaults we need
+    this.AutoRest.AddConfiguration({
+      "output-artifact": ["swagger-document.json", "swagger-document.json.map"]
+      // debug and verbose messages are not sent by default, turn them on so client settings can decide to show or not.
+      , debug: true,
+      verbose: true
+    });
+
+    // apply settings from the client
+    if (configuration) {
+      this.AutoRest.AddConfiguration(configuration);
+    }
+  }
+
+  public clear() {
+    this.artifacts.length = 0;
+  }
+}
+
+class Diagnostics {
+  // map allows us to hash the diagnostics to filter out duplicates.
+  private diagnostics = new Map<string, Diagnostic>();
+
+  public constructor(private connection: IConnection, private fileUri: string) {
+  }
+
+  public clear(send: boolean = false) {
+    this.diagnostics.clear();
+    if (send) {
+      this.send();
+    }
+  }
+
+  public send() {
+    this.connection.sendDiagnostics({ uri: this.fileUri, diagnostics: [...this.diagnostics.values()] });
+  }
+
+  public push(diagnostic: Diagnostic, send: boolean = true) {
+    const hash = md5(diagnostic) || "";
+    if (!this.diagnostics.has(hash)) {
+      this.diagnostics.set(hash, diagnostic);
+      if (send) {
+        this.send();
+      }
+    }
+  }
+}
+
+
+export interface generated {
+  messages: Array<string>;
+  files: Map<string, string>;
+}
+
+export class OpenApiLanugageService extends TextDocuments implements IFileSystem {
+  private results = new Map</*configfile*/string, Result>();
+  private diagnostics = new Map</*file*/string, Diagnostics>();
+  private virtualFile = new Map<string, TextDocument>();
+  public settings: any = {};
+
+  public get(uri: string): TextDocument {
+    const content = super.get(uri);
+    if (!content) {
+      const name = decodeURIComponent(uri);
+      for (const each of super.all()) {
+        if (decodeURIComponent(each.uri) === name) {
+          return each;
+        }
+      }
+    }
+    return content;
+  }
+
+  constructor(private connection: IConnection) {
+    super();
+
+    //  track opened, changed, saved, and closed files
+    this.onDidOpen((p) => this.onDocumentChanged(p.document));
+    this.onDidChangeContent((p) => this.onDocumentChanged(p.document));
+    this.onDidClose((p) => this.getDiagnosticCollection(p.document.uri).clear(true));
```

Comment:
are we just clearing results here? 
In the previous version of the extension, we had it behaved:
 - if config file is left opened, we do not want to clear diagnostics for referenced files
  - if config file is closed and no other referenced files are opened, then we should clear diagnostics.
why shouldn't we maintain the same behavior?